tembo-operator/src/cloudnativepg/cnpg.rs

```diff
@@ -1713,6 +1713,49 @@ fn generate_s3_restore_credentials(
     }
 }
 
+fn is_backup_completed(backup: &Backup) -> bool {
+    match &backup.status {
+        Some(status) => status.phase.as_deref() == Some("completed"),
+        None => false,
+    }
+}
+
+// Lookup the Backup status of the instance we are deploying.  If the backup isn't
+// complete, we will requeue until it is.
+pub async fn check_backups_status(cdb: &CoreDB, ctx: Arc<Context>) -> Result<(), Action> {
+    let instance_name = cdb.name_any();
+    let namespace = cdb.namespace().ok_or_else(|| {
+        error!("Namespace is not set for CoreDB for instance {}", instance_name);
+        Action::requeue(Duration::from_secs(300))
+    })?;
+
+    let backups_api: Api<Backup> = Api::namespaced(ctx.client.clone(), &namespace);
+    let label_selector = format!("cnpg.io/cluster={},cnpg.io/immediateBackup=true", instance_name);
```

Comment:
I think after 30 days, existing instances will not have any backups with immediateBackup=true, maybe we just check for any backup completed, corresponding to this instance, and completed in the last 48 hours for this step.
```
  labels:
    cnpg.io/cluster: org-***-inst-***
    cnpg.io/immediateBackup: "false"
    cnpg.io/scheduled-backup: org-***-inst-***
    ```